vnext/Scripts/msbuildeval.ps1

```diff
@@ -0,0 +1,73 @@
+# Evaluate an MSBuild property name
+param(
+    [Parameter(Mandatory=$true)]
+    [String]$SolutionFile,
+    [Parameter(Mandatory=$true)]
+    [String]$ProjectFile,
+    [Parameter(Mandatory=$true)]
+    [String]$PropertyName
+)
+
+function Get-MSBuildPath {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (!(Test-Path $vsWhere)) {
+        throw "Unable to find vswhere.exe."
+    }
+    $vsPath = & $vsWhere -version 16.5 -property installationPath;
+    return "$vsPath\MSBuild\Current\Bin\";
+}
+
+function Get-MSBuildProperty {
+    param (
+        [Parameter(Mandatory=$true)]
+        [String]$SolutionPath,
+        [Parameter(Mandatory=$true)]
+        [String]$ProjectPath,
+        [Parameter(Mandatory=$true)]
+        [String]$PropertyName
+    )
+
+    # Identify location of MSBuild
+    $MsBuildPath = Get-MSBuildPath
+
+    # Load Microsoft.Build.dll into script
+    Add-Type -Path "$MsBuildPath\Microsoft.Build.dll" | Out-Null
+
+    # Build a temporary "metaproj" of the solution file so it can be processed
+    ${env:MSBUILDEMITSOLUTION} = 1
+    & $MsBuildPath\MSBuild.exe $SolutionPath | Out-Null
+
+    # Build a local project collection
+    $projectCollection =[Microsoft.Build.Evaluation.ProjectCollection]::new()
+
+    # Process solution
+    $solution = [Microsoft.Build.Evaluation.Project]::new("$SolutionPath.metaproj", $null, "Current", $projectCollection)
+
+    # Clean up "metaproj" files
+    Remove-Item "$SolutionPath.metaproj" | Out-Null
+    Remove-Item "$SolutionPath.metaproj.tmp" | Out-Null
+
+    # Evaluate all of the Solution* properties and save into a collection
+    $globalProps = New-Object 'System.Collections.Generic.Dictionary[String,String]'
+    $solution.Properties | ForEach-Object -Process {
+        if ($_.Name.StartsWith("Solution")) {
+            $globalProps.Add($_.Name, $_.EvaluatedValue)
+        }
+    }
+
+    # Process the project file (with the Solution* properties we calculated before)
+    $project = [Microsoft.Build.Evaluation.Project]::new("$ProjectPath", $globalProps, "Current", $projectCollection)
```

Comment:
note that the global props are technically per-project (especially for platform and configuration)

Reply:
Also we should consider the environment variables as they control msbuild evaluation as well... I don't recall if the evaluatoin project required explicitly setting these as part of the projectcollection or not...

Reply:
I'm not sure. Even my processing of the solution file to extract the Solution* variables is kind of problematic, since there may be other variables I'm missing.